util/stopper.go

```diff
@@ -17,39 +17,146 @@
 
 package util
 
-import "sync"
+import (
+	"sync"
+	"sync/atomic"
+)
 
-// A Stopper provides a channel-based mechanism to stop a running
-// goroutine. Stopping occurs in two phases: the first is the request
-// to stop. The second is the confirmation by the goroutine that it
-// has stopped. Multiple goroutines can be stopped using the same
-// Stopper instance.
+const (
+	// RUNNING is the state of a stopper before invoking Stop().
+	RUNNING = int32(0)
```

Comment:
No. Forgot Go convention for a little while there. @spencerkimball I'll leave this further changes to you unless you tell me otherwise.